core/src/main/java/org/elasticsearch/script/LeafSearchScript.java

```diff
@@ -44,8 +56,23 @@ default void setNextAggregationValue(Object value) {
         setNextVar("_value", value);
     }
 
-    long runAsLong();
+    @Override
+    default void setNextVar(String field, Object value) {}
 
-    double runAsDouble();
+    /**
+     * Return the result as a long. This is used by aggregation scripts over long fields.
+     */
+    default long runAsLong() {
+        return (long) runAsDouble();
```

Comment:
I am somewhat conflicted about this. On one side I like this simplicity, on the other side if these default implementation are ever invoked, it most likely means that the script is used in a wrong context or has not been optimized for some contexts. Maybe, we shouldn't provide a default implementation, but instead clearly document in which contexts `runAsLong`, `runAsDouble` and `run` are called from and follow the same model as we do with search vs executable script and throw `UnsupportedOperationException` if a wrong method is called. I think this will encourage the users to read the comments on the method and implement the right method for their use case.